packages/stream-types-test/validation.ts

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-unused-vars, no-underscore-dangle, no-empty */
+/// <reference types="ses"/>
+
+import { makeQueue, makeStream, makePipe } from '@endo/stream';
+// eslint-disable-next-line
+import type { Stream, Reader, Writer } from '@endo/stream';
+
+async () => {
+  const q = makeQueue<number>();
+  q.put(1);
+  q.put(Promise.resolve(1));
+  const a : number = await q.get();
+}
+
+async () => {
+  const q = makeQueue<IteratorResult<number>>();
+  const r = makeQueue<IteratorResult<undefined>>();
+  const s = makeStream(q, r);
+  for await (const value of s) {}
+
+  const t = makeStream(r, q);
+  // @ts-expect-error
+  for await (const value of t) {}
+};
+
+async () => {
+  const [reader, writer] = makePipe<number>();
+  for await (const _ of reader) {}
+  // @ts-expect-error
+  for await (const _ of writer) {}
+};
+
+async () => {
+  const [reader, writer] = makePipe<undefined, number>();
+  // @ts-expect-error
```

Comment:
I'm confused why there is an expected error here actually.

Reply:
The reader and writer types are inverted. The for loop is only able to send undefined forward, but the type of reader obliges it to send numbers (and receive undefined). The second loop works.

Reply:
It just feels so awkward to read from a writer! I realize that makePipe is actually completely reversible, but it still messes with my brain, and I'm wondering if there is a use case for this, or if we shouldn't instead be opiniated and build a pipe where the `reader` always discards the `value` argument, and where the `writer` always returns `undefined` (I know it wouldn't be a simple `makeStream` with the same underlying queues anymore).